### PR TITLE
Add tutorial builder CLI 

### LINKED ISSUE #350: Feature: add CLI for creating tutorials and lessons
To further improve authors experience we could add a CLI that would automate some operations:

- generate the folder structure and the required files for each lesson
- update the `tutorials.json` file
- update the `courses.json` file

Some command proposals:

- `ps tutorial create`
- `ps lesson create`

Just a thought, let me know what you @terichadbourne

### PR DESCRIPTION
**Pull request by terichadbourne**
Work in Progress, closes #350

@zebateira I'm still working on this, but in trying to separate out the utils I've apparently created some bugs. There are times when it's successfully running a function from another file, and other times it's saying the function isn't a function. I've double-checked my export and require statements but haven't figured out yet what the problem is for some of those transitional dialogues. Mind taking a look? You can search for `BUG` to see some spots I know I'm having issues, or just work through the wizard and see where it fails. 

This is commit that causes the bugs: https://github.com/ProtoSchool/protoschool.github.io/pull/440/commits/8607a1bb2915e4d1030fa1d6af0c09a692de60b2

### CODE REVIEW

**Comment by zebateira** [comment]
@terichadbourne awesome job on this 🎉 

regarding the bug you are facing I believe it's cause by having circular dependencies. we can take a look at that later today.

File: DEVELOPING_TUTORIALS.md [hunk 1]
```diff
@@ -219,9 +302,144 @@ For example (for Lesson 01 of Tutorial 0005):
...
+Examples of meaningful resource `type`s include: `docs`, `demo`, `tutorial`, `article`, `video`, and `website`.
+
+Be sure to include any ProtoSchool tutorials that would provide a nice follow-on to your own content. Be sure to use the type `tutorial` for these, and they will automagically receive an extra callout as ProtoSchool tutorials when the page renders, like so:
+
+![screenshot](public/protoschool_resource.png)
+
+
+###### Add alert messages for new and updated tutorials (optional)
+
+TODO: Really needs to happen even if you used ProtoWizard
```
Comment (zebateira) [inline DEVELOPING_TUTORIALS.md:372]: Meaning, should we ask for a newMessage string in ProtoWizard?

**Comment by zebateira** [comment]
image

Added the following error message if user is not using at least nodejs `v10`

**Comment by zebateira** [comment]
@terichadbourne I think this is ready to go!

**Comment by zebateira** [comment]
There are still some tests that are missing for the API, but I can tackle that in another PR.

**Comment by terichadbourne** [comment]
I believe this is ready to go after a final proof from @zebateira, but it looks like the tests have turned themselves off again somehow.

**Comment by zebateira** [comment]
LGTM 🎉

**Pull request merged**